torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -0,0 +1,109 @@
+#include <cstddef>
+#include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/autodiff.h"
+
+namespace torch { namespace jit {
+
+struct Graph;
+
+namespace {
+
+// Move nodes that exist in graph g into a 'group_node_kind' node.
+// All inputs shared by the nodes become inputs to the new node.
+// Outputs from 'nodes' are redirected to outputs of the new node,
+// and the original nodes are removed.
```

Comment:
So, as an example, if I have something like
```
graph %0 where
%1 = A %0
%2 = B %1
%3 = C %2
%4 = D %3
```

and I pass `[%2], [%3]`, then I get

```
graph %0 where
%1 = A %0
%3 = Group_1 %1
%4 = D %3
with
  Group_1 %1 where
    %2 = B %1
    %3 = C %2
    ret %3
```

? Would be good to put this in the comment.